### ISSUE TITLE
These feature flags are defined in the "amplify/cli.json" configuration file and are unknown to the currently running Amplify CLI:

### ISSUE THREAD

**Issue by vladimir-virlan** [aws-amplify/amplify-hosting#2552]
### Before opening, please confirm:

- [X] I have installed the latest version of the Amplify CLI (see above), and confirmed that the issue still persists.
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-cli/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### How did you install the Amplify CLI?

npm

### If applicable, what version of Node.js are you using?

v14.18.2

### Amplify CLI Version

7.6.12

### What operating system are you using?

Windows

### Did you make any manual changes to the cloud resources managed by Amplify? Please describe the changes made.

Not really. Just added auth service and tried to deploy and keep on getting these errors. Actually I am following this guide:

https://aws.amazon.com/getting-started/hands-on/build-react-app-amplify-graphql/module-three/?e=gs2020&p=build-a-react-app-two

After that started getting the errors. Tried downgrading the cli (via npm package.json) to no avail.

### Amplify Categories

auth

### Amplify Commands

Not applicable

### Describe the bug

The error happens when I push latest changes to github and the build triggers...

### Expected behavior

Build to complete on push successfully..

### Reproduction steps

1. Follow this guide (or just add auth) and try to deploy by pushing to github: https://aws.amazon.com/getting-started/hands-on/build-react-app-amplify-graphql/module-three/?e=gs2020&p=build-a-react-app-two
2. Make sure github is linked to your amplify and make sure frontend is connected to backend:
![image](https://user-images.githubusercontent.com/89448293/151238295-296fbf83-3c0f-4672-8483-a909a97dfa31.png)
3. When you push the changes, provision is OK, but Build fails with the specificed error.
Thanks 

### GraphQL schema(s)

<details>

```graphql
# Put schemas below this line


```

</details>

### Log output

<details>

```
# Put your logs below this line
                                # Starting phase: build
2022-01-26T19:51:53.229Z [INFO]: [31mInvalid feature flag configuration[39m
2022-01-26T19:51:53.231Z [INFO]: [31mThese feature flags are defined in the "amplify/cli.json" configuration file and are unknown to the currently running Amplify CLI:[39m
                                 [31m  - project[39m
                                 [31m  - graphqltransformer.transformerversion[39m
                                 [31m  - graphqltransformer.suppressschemamigrationprompt[39m
                                 [31m  - graphqltransformer.securityenhancementnotification[39m
                                 [31m  - auth.useenabledmfas[39m
                                 [31mThis issue likely happens when the project has been pushed with a newer version of Amplify CLI, try updating to a newer version.[39m
                                 [31mEnsure that the CI/CD pipeline is not using an older or pinned down version of Amplify CLI.[39m
                                 [31mLearn more about feature flags: https://docs.amplify.aws/cli/reference/feature-flags[39m
2022-01-26T19:51:53.243Z [ERROR]: !!! Build failed
2022-01-26T19:51:53.243Z [ERROR]: !!! Non-Zero Exit Code detected
2022-01-26T19:51:53.243Z [INFO]: # Starting environment caching...
2022-01-26T19:51:53.244Z [INFO]: # Uploading environment cache artifact...
2022-01-26T19:51:53.317Z [INFO]: # Environment caching completed
Terminating logging...

```

</details>

**Comment by jhockett** [comment]
@vladimir-virlan glad to hear you're unblocked! The issue has been transferred so that the Console team can look into why you weren't getting the `@latest` Amplify CLI version by default.

**Comment by jakejcheng** [comment]
> @vladimir-virlan the Console always updates to the version in your build settings, which defaults to @latest of Amplify CLI, so the Console isn't actually using 6.3.1.

holy moly I spent two hours adding/removing config files, re-pulling amplify, redeploying commits and researching this issue and eventually found this https://github.com/aws-amplify/amplify-cli/issues/9498. But it was still no help and finally forcing the Amplify CLI version in build settings works!

So how this issue was triggered for me was:
- Everything was fine until 4 hours ago (last frontend deployment was 20 hours ago)
- I connected another branch through the console (I only had 1 branch/frontend env)
- I'm pretty sure I checked "Enable full-stack continuous deployments (CI/CD)"
- Branched out from the main branch that was used to deploy 20 hours ago (so it should work)
- Then during backend build phase, it failed with the error above
- Till this point, I have not updated amplify cli, aws cli, modified any amplify settings, redeployed frontend/backend envs to the original env/branch through the console
- There's definitely a bug somewhere that "downgrades" the CLI version upon connecting a branch

**Comment by amitvchaudhary** [comment]
Facing same issue while checking out the branch. Below is the error message.

Invalid feature flag configuration

These feature flags are defined in the "amplify/cli.json" configuration file and are unknown to the currently running Amplify CLI:
  - graphqltransformer.showfieldauthnotification

This issue likely happens when the project has been pushed with a newer version of Amplify CLI, try updating to a newer version.

Learn more about feature flags: https://docs.amplify.aws/cli/reference/feature-flags

**Comment by duyetphamm** [comment]
> Facing same issue while checking out the branch. Below is the error message. It was working fine till day before yesterday.
> 
> Invalid feature flag configuration
> 
> These feature flags are defined in the "amplify/cli.json" configuration file and are unknown to the currently running Amplify CLI:
> 
> * graphqltransformer.showfieldauthnotification
> 
> This issue likely happens when the project has been pushed with a newer version of Amplify CLI, try updating to a newer version.
> 
> Learn more about feature flags: https://docs.amplify.aws/cli/reference/feature-flags

I got the same issue. Please help

**Comment by duyetphamm** [comment]
> Facing same issue while checking out the branch. Below is the error message. It was working fine till day before yesterday.
> 
> Invalid feature flag configuration
> 
> These feature flags are defined in the "amplify/cli.json" configuration file and are unknown to the currently running Amplify CLI:
> 
> * graphqltransformer.showfieldauthnotification
> 
> This issue likely happens when the project has been pushed with a newer version of Amplify CLI, try updating to a newer version.
> 
> Learn more about feature flags: https://docs.amplify.aws/cli/reference/feature-flags

you should update version of Amplify CLI first. Go to source folder and run npm i -g @aws-amplify/cli @amitvchaudhary

**Comment by Abdeluswho** [comment]
Adding amplify-cli to the build setting fixed the issue for me. Thanks

**Comment by gawshan** [comment]
@Abdeluswho , where did you add it? I am still having the issue.

**Comment by Abdeluswho** [comment]
> @Abdeluswho , where did you add it? I am still having the issue.

In the Amplify UI if you go to build settings and scroll all the way down there is a section for packages. Add it there and set it to latest. That should work

**Comment by martinburger** [comment]
It seems I've stumbled upon the same and a similar issue, respectively.

On my local machine, I am using version `7.6.21` of the `@aws-amplify/cli` package.

The backend build fails due to the following errors:

```
                                 ## Checking for associated backend environment...
                                 ## Backend environment association found
2022-02-21T16:11:03.570Z [INFO]: [31mInvalid feature flag configuration[39m
2022-02-21T16:11:03.570Z [INFO]: [31mThese feature flags are defined in the "amplify/cli.json" configuration file and are unknown to the currently running Amplify CLI:[39m
                                 [31m  - project[39m
                                 [31m  - graphqltransformer.transformerversion[39m
                                 [31m  - graphqltransformer.suppressschemamigrationprompt[39m
                                 [31m  - graphqltransformer.securityenhancementnotification[39m
                                 [31m  - graphqltransformer.showfieldauthnotification[39m
                                 [31m  - auth.useenabledmfas[39m
2022-02-21T16:11:03.571Z [INFO]: [31mThis issue likely happens when the project has been pushed with a newer version of Amplify CLI, try updating to a newer version.[39m
                                 [31mEnsure that the CI/CD pipeline is not using an older or pinned down version of Amplify CLI.[39m
2022-02-21T16:11:03.571Z [INFO]: [31mLearn more about feature flags: https://docs.amplify.aws/cli/reference/feature-flags[39m
2022-02-21T16:11:03.581Z [ERROR]: !!! Build failed
2022-02-21T16:11:03.581Z [ERROR]: !!! Non-Zero Exit Code detected
2022-02-21T16:11:03.581Z [INFO]: # Starting environment caching...
2022-02-21T16:11:03.581Z [INFO]: # Uploading environment cache artifact...
2022-02-21T16:11:03.674Z [INFO]: # Environment caching completed
Terminating logging...
```

To fix this, I configured the build image settings to override the default installed version of package Amplify CLI to above-mentioned version `7.6.21`.

Now, it seems that Amplify fails to patch NPM package `@aws-amplify/cli` to `7.6.21`:

```
2022-02-21T16:23:29.471Z [INFO]: # Node version 17 is available for installation
2022-02-21T16:23:29.542Z [INFO]: # Installing Node version 17
2022-02-21T16:23:35.510Z [INFO]: # Now using Node version 17
2022-02-21T16:23:36.282Z [WARNING]: Cannot find module '/root/.nvm/versions/node/v17.5.0/lib/node_modules/@aws-amplify/cli/package.json'
                                    Require stack:
                                    - /snapshot/AWSMobilePioneerExecute/src/plugins/patch/npm.js
                                    - /snapshot/AWSMobilePioneerExecute/src/plugins/patch/patch.js
                                    - /snapshot/AWSMobilePioneerExecute/pioneer.js
2022-02-21T16:23:36.282Z [INFO]: # Patching NPM package '@aws-amplify/cli' to 7.6.21...
2022-02-21T16:24:03.005Z [INFO]: npm WARN ERESOLVE overriding peer dependency
2022-02-21T16:25:25.324Z [INFO]: npm WARN deprecated source-map-url@0.4.1: See https://github.com/lydell/source-map-url#deprecated
2022-02-21T16:25:25.801Z [INFO]: npm
2022-02-21T16:25:25.802Z [INFO]: 
2022-02-21T16:25:25.802Z [INFO]: WARN
2022-02-21T16:25:25.802Z [INFO]: 
2022-02-21T16:25:25.803Z [INFO]: deprecated
2022-02-21T16:25:25.803Z [INFO]: urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
2022-02-21T16:25:26.641Z [INFO]: npm
2022-02-21T16:25:26.641Z [INFO]: WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
2022-02-21T16:25:26.970Z [INFO]: npm WARN deprecated
2022-02-21T16:25:26.971Z [INFO]: source-map-resolve@0.5.3: See https://github.com/lydell/source-map-resolve#deprecated
2022-02-21T16:25:27.634Z [INFO]: npm
2022-02-21T16:25:27.634Z [INFO]: WARN deprecated querystring@0.2.0: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.
2022-02-21T16:25:27.732Z [INFO]: npm
2022-02-21T16:25:27.733Z [INFO]: WARN deprecated graphql-import@0.7.1: GraphQL Import has been deprecated and merged into GraphQL Tools, so it will no longer get updates. Use GraphQL Tools instead to stay up-to-date! Check out https://www.graphql-tools.com/docs/migration-from-import for migration and https://the-guild.dev/blog/graphql-tools-v6 for new changes.
2022-02-21T16:25:28.116Z [INFO]: npm WARN
2022-02-21T16:25:28.116Z [INFO]: deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
2022-02-21T16:25:28.118Z [INFO]: npm
2022-02-21T16:25:28.118Z [INFO]: WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
2022-02-21T16:25:28.120Z [INFO]: npm
2022-02-21T16:25:28.120Z [INFO]: WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
2022-02-21T16:25:28.300Z [INFO]: npm WARN
2022-02-21T16:25:28.302Z [INFO]: deprecated @graphql-toolkit/common@0.9.7: GraphQL Toolkit is deprecated and merged into GraphQL Tools, so it will no longer get updates. Use GraphQL Tools instead to stay up-to-date! Check out https://www.graphql-tools.com/docs/migration-from-toolkit for migration and https://the-guild.dev/blog/graphql-tools-v6 for new changes.
2022-02-21T16:25:28.303Z [INFO]: npm
2022-02-21T16:25:28.303Z [INFO]: WARN deprecated @graphql-toolkit/common@0.9.7: GraphQL Toolkit is deprecated and merged into GraphQL Tools, so it will no longer get updates. Use GraphQL Tools instead to stay up-to-date! Check out https://www.graphql-tools.com/docs/migration-from-toolkit for migration and https://the-guild.dev/blog/graphql-tools-v6 for new changes.
2022-02-21T16:25:28.353Z [INFO]: npm WARN
2022-02-21T16:25:28.353Z [INFO]: deprecated uuid@3.3.2: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
2022-02-21T16:25:30.232Z [INFO]: npm WARN
2022-02-21T16:25:30.234Z [INFO]: deprecated @graphql-toolkit/common@0.6.6: GraphQL Toolkit is deprecated and merged into GraphQL Tools, so it will no longer get updates. Use GraphQL Tools instead to stay up-to-date! Check out https://www.graphql-tools.com/docs/migration-from-toolkit for migration and https://the-guild.dev/blog/graphql-tools-v6 for new changes.
2022-02-21T16:25:43.764Z [INFO]: npm WARN deprecated core-js@2.6.12: core-js@<3.4 is no longer maintained and not recommended for usage due to the number of issues. Because of the V8 engine whims, feature detection in old core-js versions could cause a slowdown up to 100x even if nothing is polyfilled. Please, upgrade your dependencies to the actual version of core-js.
2022-02-21T16:25:43.765Z [INFO]: npm WARN deprecated core-js@2.6.12: core-js@<3.4 is no longer maintained and not recommended for usage due to the number of issues. Because of the V8 engine whims, feature detection in old core-js versions could cause a slowdown up to 100x even if nothing is polyfilled. Please, upgrade your dependencies to the actual version of core-js.
                                 npm
2022-02-21T16:25:43.765Z [INFO]: WARN deprecated core-js@2.6.12: core-js@<3.4 is no longer maintained and not recommended for usage due to the number of issues. Because of the V8 engine whims, feature detection in old core-js versions could cause a slowdown up to 100x even if nothing is polyfilled. Please, upgrade your dependencies to the actual version of core-js.
2022-02-21T16:25:49.467Z [INFO]: npm ERR! code 127
2022-02-21T16:25:49.469Z [INFO]: npm ERR! path /root/.nvm/versions/node/v17.5.0/lib/node_modules/@aws-amplify/cli/node_modules/babel-runtime/node_modules/core-js
                                 npm ERR! command failed
                                 npm ERR! command sh -c node -e "try{require('./postinstall')}catch(e){}"
                                 npm ERR!
2022-02-21T16:25:49.469Z [INFO]: sh: node: command not found
2022-02-21T16:25:49.469Z [INFO]: 
2022-02-21T16:25:49.470Z [INFO]: npm
2022-02-21T16:25:49.470Z [INFO]: ERR! A complete log of this run can be found in:
                                 npm ERR!     /root/.npm/_logs/2022-02-21T16_23_37_550Z-debug-0.log
2022-02-21T16:25:49.526Z [INFO]: ! Error patching NPM package '@aws-amplify/cli' to version 7.6.21
2022-02-21T16:25:49.538Z [INFO]: # Retrieving cache...
2022-02-21T16:25:51.925Z [INFO]: # Extracting cache...
2022-02-21T16:25:54.423Z [INFO]: # Extraction completed
```

As a consequence, the backend build still fails with the same errors:

```
                                 # Starting phase: build
2022-02-21T16:26:18.327Z [INFO]: [31mInvalid feature flag configuration[39m
2022-02-21T16:26:18.328Z [INFO]: [31mThese feature flags are defined in the "amplify/cli.json" configuration file and are unknown to the currently running Amplify CLI:[39m
                                 [31m  - project[39m
                                 [31m  - graphqltransformer.transformerversion[39m
                                 [31m  - graphqltransformer.suppressschemamigrationprompt[39m
                                 [31m  - graphqltransformer.securityenhancementnotification[39m
                                 [31m  - graphqltransformer.showfieldauthnotification[39m
                                 [31m  - auth.useenabledmfas[39m
                                 [31mThis issue likely happens when the project has been pushed with a newer version of Amplify CLI, try updating to a newer version.[39m
2022-02-21T16:26:18.328Z [INFO]: [31mEnsure that the CI/CD pipeline is not using an older or pinned down version of Amplify CLI.[39m
                                 [31mLearn more about feature flags: https://docs.amplify.aws/cli/reference/feature-flags[39m
2022-02-21T16:26:18.339Z [ERROR]: !!! Build failed
2022-02-21T16:26:18.340Z [ERROR]: !!! Non-Zero Exit Code detected
2022-02-21T16:26:18.340Z [INFO]: # Starting environment caching...
2022-02-21T16:26:18.340Z [INFO]: # Uploading environment cache artifact...
2022-02-21T16:26:18.408Z [INFO]: # Environment caching completed
```

Any ideas how to fix this?

**Comment by martinburger** [comment]
Same issue with Node.js 16:

```
2022-02-22T06:22:07.518Z [INFO]: # Node version 16 is available for installation
2022-02-22T06:22:07.519Z [INFO]: # Installing Node version 16
2022-02-22T06:22:12.729Z [INFO]: # Now using Node version 16
2022-02-22T06:22:13.439Z [WARNING]: Cannot find module '/root/.nvm/versions/node/v16.14.0/lib/node_modules/@aws-amplify/cli/package.json'
                                    Require stack:
                                    - /snapshot/AWSMobilePioneerExecute/src/plugins/patch/npm.js
                                    - /snapshot/AWSMobilePioneerExecute/src/plugins/patch/patch.js
                                    - /snapshot/AWSMobilePioneerExecute/pioneer.js
2022-02-22T06:22:13.439Z [INFO]: # Patching NPM package '@aws-amplify/cli' to 7.6.13...
2022-02-22T06:22:45.797Z [INFO]: npm WARN ERESOLVE overriding peer dependency
2022-02-22T06:24:08.684Z [INFO]: npm
2022-02-22T06:24:08.686Z [INFO]: WARN deprecated source-map-url@0.4.1: See https://github.com/lydell/source-map-url#deprecated
2022-02-22T06:24:09.015Z [INFO]: npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
2022-02-22T06:24:09.192Z [INFO]: npm
2022-02-22T06:24:09.192Z [INFO]: WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
2022-02-22T06:24:09.517Z [INFO]: npm WARN
2022-02-22T06:24:09.518Z [INFO]: deprecated source-map-resolve@0.5.3: See https://github.com/lydell/source-map-resolve#deprecated
2022-02-22T06:24:10.416Z [INFO]: npm WARN
2022-02-22T06:24:10.416Z [INFO]: deprecated querystring@0.2.0: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.
2022-02-22T06:24:10.505Z [INFO]: npm WARN
2022-02-22T06:24:10.506Z [INFO]: deprecated graphql-import@0.7.1: GraphQL Import has been deprecated and merged into GraphQL Tools, so it will no longer get updates. Use GraphQL Tools instead to stay up-to-date! Check out https://www.graphql-tools.com/docs/migration-from-import for migration and https://the-guild.dev/blog/graphql-tools-v6 for new changes.
2022-02-22T06:24:10.853Z [INFO]: npm WARN
2022-02-22T06:24:10.854Z [INFO]: deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
2022-02-22T06:24:10.855Z [INFO]: npm
2022-02-22T06:24:10.855Z [INFO]: WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
2022-02-22T06:24:10.858Z [INFO]: npm
2022-02-22T06:24:10.858Z [INFO]: WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
2022-02-22T06:24:10.930Z [INFO]: npm
2022-02-22T06:24:10.930Z [INFO]: WARN deprecated @graphql-toolkit/common@0.9.7: GraphQL Toolkit is deprecated and merged into GraphQL Tools, so it will no longer get updates. Use GraphQL Tools instead to stay up-to-date! Check out https://www.graphql-tools.com/docs/migration-from-toolkit for migration and https://the-guild.dev/blog/graphql-tools-v6 for new changes.
2022-02-22T06:24:10.941Z [INFO]: npm WARN
2022-02-22T06:24:10.942Z [INFO]: deprecated @graphql-toolkit/common@0.9.7: GraphQL Toolkit is deprecated and merged into GraphQL Tools, so it will no longer get updates. Use GraphQL Tools instead to stay up-to-date! Check out https://www.graphql-tools.com/docs/migration-from-toolkit for migration and https://the-guild.dev/blog/graphql-tools-v6 for new changes.
2022-02-22T06:24:10.984Z [INFO]: npm WARN
2022-02-22T06:24:10.984Z [INFO]: deprecated uuid@3.3.2: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
2022-02-22T06:24:13.589Z [INFO]: npm
2022-02-22T06:24:13.591Z [INFO]: WARN deprecated @graphql-toolkit/common@0.6.6: GraphQL Toolkit is deprecated and merged into GraphQL Tools, so it will no longer get updates. Use GraphQL Tools instead to stay up-to-date! Check out https://www.graphql-tools.com/docs/migration-from-toolkit for migration and https://the-guild.dev/blog/graphql-tools-v6 for new changes.
2022-02-22T06:24:26.289Z [INFO]: npm WARN deprecated @aws-amplify/cli@7.6.13: This version is no longer supported. Please use 7.6.15 or higher.
2022-02-22T06:24:27.452Z [INFO]: npm WARN deprecated core-js@2.6.12: core-js@<3.4 is no longer maintained and not recommended for usage due to the number of issues. Because of the V8 engine whims, feature detection in old core-js versions could cause a slowdown up to 100x even if nothing is polyfilled. Please, upgrade your dependencies to the actual version of core-js.
2022-02-22T06:24:27.453Z [INFO]: npm
2022-02-22T06:24:27.453Z [INFO]: WARN deprecated core-js@2.6.12: core-js@<3.4 is no longer maintained and not recommended for usage due to the number of issues. Because of the V8 engine whims, feature detection in old core-js versions could cause a slowdown up to 100x even if nothing is polyfilled. Please, upgrade your dependencies to the actual version of core-js.
2022-02-22T06:24:27.454Z [INFO]: npm
2022-02-22T06:24:27.454Z [INFO]: WARN deprecated core-js@2.6.12: core-js@<3.4 is no longer maintained and not recommended for usage due to the number of issues. Because of the V8 engine whims, feature detection in old core-js versions could cause a slowdown up to 100x even if nothing is polyfilled. Please, upgrade your dependencies to the actual version of core-js.
2022-02-22T06:24:33.552Z [INFO]: npm
2022-02-22T06:24:33.556Z [INFO]: ERR! code 127
                                 npm ERR! path /root/.nvm/versions/node/v16.14.0/lib/node_modules/@aws-amplify/cli/node_modules/babel-runtime/node_modules/core-js
                                 npm ERR! command failed
                                 npm ERR! command sh -c node -e "try{require('./postinstall')}catch(e){}"
                                 npm ERR! sh: node: command not found
                                 npm ERR! A complete log of this run can be found in:
                                 npm ERR!     /root/.npm/_logs/2022-02-22T06_22_14_753Z-debug-0.log
2022-02-22T06:24:33.611Z [INFO]: ! Error patching NPM package '@aws-amplify/cli' to version 7.6.13
2022-02-22T06:24:33.623Z [INFO]: # Retrieving cache...
2022-02-22T06:24:45.684Z [INFO]: # Extracting cache...
2022-02-22T06:24:48.255Z [INFO]: # Extraction completed
```

**Comment by oste** [comment]
I am also facing this issue. Currently looking at using a custom image to rule out if maybe something is off with the live update to latest. Interestingly, if I work back from version 7.6.21 it eventually started to fail at a `amplify --version` instead of `amplifyPush --simple`. I tried all versions of node as well with no luck.

**Comment by martinburger** [comment]
> I am also facing this issue. Currently looking at using a custom image to rule out if maybe something is off with the live update to latest. Interestingly, if I work back from version 7.6.21 it eventually started to fail at a `amplify --version` instead of `amplifyPush --simple` -- with the same feature flags config error. I tried all versions of node as well with no luck.

You find sections `backend` and `frontend`  of my applications's configuration file below. I added some commands to verify the Amplify and the Node.js version, respectively.

```yaml
version: 1
backend:
  phases:
    preBuild:
      commands:
        - node --version
        - amplify --version
    build:
      commands:
        - amplifyPush --simple
frontend:
  phases:
    preBuild:
      commands:
        - node --version
        - npm ci
    build:
      commands:
        - npm run build
  artifacts:
    baseDirectory: build
    files:
      - '**/*'
  cache:
    paths:
      - node_modules/**/*
```

Now, the backend build fails already at command `amplify --version` in phase `preBuild`:

```
                                 # Starting phase: preBuild
                                 # Executing command: node --version
2022-02-22T13:25:56.374Z [INFO]: v17.5.0
                                 # Executing command: amplify --version
2022-02-22T13:25:59.450Z [INFO]: [31mInvalid feature flag configuration[39m
[...]
```

**Comment by martinburger** [comment]
> can you try doing a clean install if you're not doing it already? using `npm ci` instead of `npm install` in prebuild stage.

Thanks for the hint. I use `npm` in the frontend section only:

```yaml
version: 1
backend:
  phases:
    preBuild:
      commands:
        - node --version
        - amplify --version
    build:
      commands:
        - amplifyPush --simple
frontend:
  phases:
    preBuild:
      commands:
        - node --version
        - npm ci
# [...]
```

And, I already use `npm ci` to install packages.

**Comment by martinburger** [comment]
> also, you can try resetting [env cache](https://github.com/aws-amplify/amplify-console/issues/229), idk if it helps but something better than nothing right?

I did that as well:

```yaml
version: 1
backend:
  phases:
    preBuild:
      commands:
        # https://github.com/aws-amplify/amplify-console/issues/2552#issuecomment-1048038009
        - envCache --set stackInfo ""
        - node --version
        - amplify --version
    build:
      commands:
        - amplifyPush --simple
# [...]
```

To no avail as well:

```
                                 # Starting phase: preBuild
                                 # Executing command: envCache --set stackInfo ""
2022-02-23T06:31:46.827Z [INFO]: # Executing command: node --version
2022-02-23T06:31:46.830Z [INFO]: v17.6.0
2022-02-23T06:31:46.830Z [INFO]: # Executing command: amplify --version
2022-02-23T06:31:49.972Z [INFO]: [31mInvalid feature flag configuration[39m
```

**Comment by martinburger** [comment]
I just tried to update global package `@aws-amplify/cli` in my build script:

```yaml
version: 1
backend:
  phases:
    preBuild:
      commands:
        - node --version
        - npm update -g @aws-amplify/cli
        - amplify --version
    build:
      commands:
        - amplifyPush --simple
# [...]
```
Both, with overriding Amplify CLI to latest and without overriding that package, same result:

```
                                 # Starting phase: preBuild
                                 # Executing command: node --version
2022-02-23T06:47:12.903Z [INFO]: v17.6.0
                                 # Executing command: npm update -g @aws-amplify/cli
2022-02-23T06:47:15.125Z [INFO]: changed 1 package, and audited 220 packages in 2s
2022-02-23T06:47:15.125Z [INFO]: 11 packages are looking for funding
                                 run `npm fund` for details
2022-02-23T06:47:15.127Z [INFO]: 3 moderate severity vulnerabilities
                                 To address all issues, run:
                                 npm audit fix
                                 Run `npm audit` for details.
2022-02-23T06:47:15.133Z [INFO]: # Executing command: amplify --version
2022-02-23T06:47:18.307Z [INFO]: [31mInvalid feature flag configuration[39m
```

So, although it seems that package `@aws-amplify/cli` had been updated, the backend build still failed with the same `Invalid feature flag configuration` error. Well, I've run out of ideas.

BTW, without overriding Amplify CLI, the `Cannot find module ...` error while cloning the repository doesn't occur.

**Comment by donalbertico** [comment]
updating amplify cli helped for me `sudo npm i -g @aws-amplify/cli`

**Comment by martinburger** [comment]
> updating amplify cli helped for me `sudo npm i -g @aws-amplify/cli`

In my case, using `sudo` causes the following error:

```
                                 # Starting phase: preBuild
                                 # Executing command: sudo npm install -g @aws-amplify/cli
2022-02-23T15:07:08.231Z [WARNING]: /codebuild/output/src803801990/src/erlebe-salsa-website/amplify.sh: line 5: sudo: command not found
2022-02-23T15:07:08.231Z [ERROR]: !!! Build failed
2022-02-23T15:07:08.231Z [ERROR]: !!! Non-Zero Exit Code detected
2022-02-23T15:07:08.232Z [INFO]: # Starting environment caching...
2022-02-23T15:07:08.232Z [INFO]: # Uploading environment cache artifact...
2022-02-23T15:07:08.334Z [INFO]: # Environment caching completed
Terminating logging...
```

**Comment by ritika2291** [comment]
I'm also facing the same issue, using version 7.6.21.

**Comment by gtondello** [comment]
For me, in addition to setting the Amplify CLI version to the latest in the build settings, I had to edit the App details in the General settings and select a Service Role. After that, it worked.

Thank you for this thread, it was really helpful so I could find a solution quickly.

**Comment by Ajit-Singh-Brar** [comment]
Hi everyone, I am facing the same issue. Can you guys help me out as my cli/json file looks like this:
```
{
  "features": `{`
    "graphqltransformer": {
      "addmissingownerfields": true,
      "securityEnhancementNotification": false,
      "showfieldauthnotification": false
    },
    "project": {
      "overrides": true
    }
  }
}
```

Can you help me with this as it shows the 
`                                 # Starting phase: build
2022-03-09T17:41:19.266Z [INFO]: [31mInvalid feature flag configuration[39m
2022-03-09T17:41:19.267Z [INFO]: [31mThese feature flags are defined in the "amplify/cli.json" configuration file and are unknown to the currently running Amplify CLI:[39m
                                 [31m  - project[39m
                                 [31m  - graphqltransformer.securityenhancementnotification[39m
                                 [31m  - graphqltransformer.showfieldauthnotification[39m
                                 [31mThis issue likely happens when the project has been pushed with a newer version of Amplify CLI, try updating to a newer version.[39m
                                 [31mEnsure that the CI/CD pipeline is not using an older or pinned down version of Amplify CLI.[39m
                                 [31mLearn more about feature flags: https://docs.amplify.aws/cli/reference/feature-flags[39m
2022-03-09T17:41:19.276Z [ERROR]: !!! Build failed
2022-03-09T17:41:19.276Z [ERROR]: !!! Non-Zero Exit Code detected
2022-03-09T17:41:19.276Z [INFO]: # Starting environment caching...
2022-03-09T17:41:19.277Z [INFO]: # Uploading environment cache artifact...
2022-03-09T17:41:19.354Z [INFO]: # Environment caching completed
Terminating logging...`

**Comment by Ajit-Singh-Brar** [comment]
I want to understand which place I can update the CLI to make my project work

**Comment by sethjeffery** [comment]
Setting Amplify CLI to latest in the build image settings worked for me.

<img width="626" alt="image" src="https://user-images.githubusercontent.com/1628558/157634345-23ebbf0d-273a-47f9-8adb-06019ca199f5.png">

- Click **Build settings** from the left
- Click **Edit** button in the Build Image settings box
- Click **Add package version override** and select **Amplify CLI**
- Save changes and rebuild
- Annoying "These feature flags" message went away for me.

**Comment by junglesquirrel** [comment]
Hey all, to further clarify the working fix - the above instructions from @sethjeffery combined with assigning a service role to the application worked for me (also described by @gtondello).

**Comment by Michael-Rimmer** [comment]
> Hey all, to further clarify the working fix - the above instructions from @sethjeffery combined with assigning a service role to the application worked for me (also described by @gtondello).

As well as this, I had to run 'amplify upgrade' on my local development device.

**Comment by cdeanhardt** [comment]
like @gtondello experienced, I had to do both
* set the Amplify CLI version to the latest in build settings 
* create/choose a service role in App Details
This is the special sauce.  

I'm very new to Amplify and was going through the "getting started, hands on, graphql" tutorial for beginners when this tripped me up. https://aws.amazon.com/getting-started/hands-on/build-react-app-amplify-graphql 

> For me, in addition to setting the Amplify CLI version to the latest in the build settings, I had to edit the App details in the General settings and select a Service Role. After that, it worked.
> 
> Thank you for this thread, it was really helpful so I could find a solution quickly.

**Comment by bsikute** [comment]
> like @gtondello experienced, I had to do both
> 
> * set the Amplify CLI version to the latest in build settings
> * create/choose a service role in App Details
> 
> This is the special sauce.
> 
> I'm very new to Amplify and was going through the "getting started, hands on, graphql" tutorial for beginners when this tripped me up. https://aws.amazon.com/getting-started/hands-on/build-react-app-amplify-graphql
> 
> > For me, in addition to setting the Amplify CLI version to the latest in the build settings, I had to edit the App details in the General settings and select a Service Role. After that, it worked.
> > Thank you for this thread, it was really helpful so I could find a solution quickly.

Thanks, this worked for me.. am good now

**Comment by R-Art-project** [comment]
> Setting Amplify CLI to latest in the build image settings worked for me.
> 
> <img alt="image" width="626" src="https://user-images.githubusercontent.com/1628558/157634345-23ebbf0d-273a-47f9-8adb-06019ca199f5.png">
> 
> * Click **Build settings** from the left
> * Click **Edit** button in the Build Image settings box
> * Click **Add package version override** and select **Amplify CLI**
> * Save changes and rebuild
> * Annoying "These feature flags" message went away for me.

Hi @sethjeffery this work for me, thank!!!

**Comment by vivian-usc-2009** [comment]
> frontend:
>   phases:
>     preBuild:
>       commands:
>         - npm ci
>         - nvm use $VERSION_NODE_16
>         - npm install aws-amplify
>         - amplify --version
>         - npm install --silent --global expo-cli
>         - |-
>           if [ -f yarn.lock ]; then
>            yarn
>           elif [ -f package-lock.json ] || [ -f npm-shrinkwrap.json ]; then
>            npm ci
>           else
>            npm install
>           fi
>           
>     build:
>       commands:
>         - nvm use $VERSION_NODE_16
>         - 'expo build:web'

Thank you! This solves my problem.

**Comment by vivian-usc-2009** [comment]
> Setting Amplify CLI to latest in the build image settings worked for me.
> 
> <img alt="image" width="626" src="https://user-images.githubusercontent.com/1628558/157634345-23ebbf0d-273a-47f9-8adb-06019ca199f5.png">
> 
> * Click **Build settings** from the left
> * Click **Edit** button in the Build Image settings box
> * Click **Add package version override** and select **Amplify CLI**
> * Save changes and rebuild
> * Annoying "These feature flags" message went away for me.


Thank you! This solves my problem.

**Comment by ANRCorleone** [comment]
I was following [this](https://aws.amazon.com/getting-started/hands-on/build-react-app-amplify-graphql/module-five/?e=gs2020&p=build-a-react-app-four) intro tutorial
and got blocked by a related issue. The other outdated stuff in the tutorial were pretty easy to fix, this one took the longest to figure out. Would be nice if the live package updates defaults to the latest version, so we don't have to do the workaround here. I'm currently at amplify cli 9.2.1 locally, the remote seems to be defaulting to an older version

`2022-08-23T03:42:44.482Z [INFO]: # Patching NPM package '@aws-amplify/cli' from 7.6.14 to 9.2.1...
2022-08-23T03:44:09.386Z [INFO]: # Done patching NPM package '@aws-amplify/cli' to version 9.2.1`

**Comment by makyfj** [comment]
The solution I has was the following: 
Current node version for my project is 14.15.0
amplify.yml
```yml
version: 1
backend:
  phases:
    preBuild:
      commands:
        - nvm install 14.15.0
        - nvm use 14.15.0
        - npm install -g yarn
        - npm install -g @aws-amplify/cli
        - yarn
    build:
      commands:
        - echo Branch is ${AWS_BRANCH} 
        - amplifyPush --simple
frontend:
  phases:
    preBuild:
      commands:
        - yarn install
    build:
      commands:
        - yarn run build
        - REACT_APP_ENV=${REACT_APP_ENV}
        - REACT_APP_SQUARE_CDN=${REACT_APP_SQUARE_CDN}
  artifacts:
    baseDirectory: build
    files:
      - '**/*'
 ```
And for build image settings: I chose version latest

**Comment by iwasinnam** [comment]
A year has passed. This is still happening. Just happened to me. Only when I followed the suggestion above and explicitly wrote Amplify CLI version 10.8.1 instead of "latest" under Build Settings did it start working again.

**Comment by swaminator** [comment]
By design, the console uses the latest version of the Amplify CLI in the build so you always get the latest capabilities in your build machine. Having said that some teams decide to fix to a version of the CLI - in this case it is also expected to update your build settings to pin the CLI version so there are no mismatches between your dev machine and the CI/CD container. Resolving this issue.
